src/main/java/com/arcanewarrior/disguises/events/DisguiseEvents.java

```diff
@@ -81,14 +85,80 @@ private void handlePlayerPacketInput(PlayerPacketEvent event) {
     }
 
     private void playerEntityAnimation(PlayerPacketOutEvent event) {
-        if(event.getPacket() instanceof EntityAnimationPacket packet) {
-            Entity disguise = parentManager.getPlayerDisguise(event.getPlayer());
-            if(disguise != null) {
+        if (event.getPacket() instanceof EntityAnimationPacket packet) {
+            Disguise disguise = parentManager.getPlayerDisguise(event.getPlayer());
+            if (disguise != null) {
                 // Translate animation from player to disguise
                 // TODO: Check to see which animations may not need translation, may have to check per entity type
                 EntityAnimationPacket newPacket = new EntityAnimationPacket(disguise.getEntityId(), packet.animation());
                 PacketUtils.sendGroupedPacket(disguise.getViewers(), newPacket);
             }
         }
     }
+
+    private void playerTeleport(PlayerPacketOutEvent event) {
+        if (event.getPacket() instanceof EntityTeleportPacket packet) {
+            Disguise disguise = parentManager.getPlayerDisguise(event.getPlayer());
+            if (disguise != null) {
+                EntityTeleportPacket newPacket = new EntityTeleportPacket(disguise.getEntityId(), packet.position(), packet.onGround());
+                PacketUtils.sendGroupedPacket(disguise.getViewers(), newPacket);
+            }
+        }
+    }
+
+    private void playerSpawn(PlayerPacketOutEvent event) {
```

Comment:
Not sure what this is intended to do, reads to me as when the player spawns and is disguises, removes themselves from viewing themself and re displays disguise?

If this is attempting to show all active disguises to a newly joined player, that's not what this appears to do.

Additionally, I would probably use a PlayerSpawnEvent listener instead of listening to all outgoing packets.

Reply:
I will do it later